scripts/lib/free_port.sh

```
@@ -0,0 +1,69 @@
+    pids=$(lsof -ti "tcp:${port}" -sTCP:LISTEN 2>/dev/null || true)
+  fi
+
+  if [[ -z "${pids}" ]]; then
+    return 0
+  fi
+
+  echo "Port ${port} in use — stopping PID(s): ${pids} (${label})..."
+  kill ${pids} 2>/dev/null || true
+  sleep 1
+  if ss -tlnp "sport = :${port}" >/dev/null 2>&1; then
+    kill -9 ${pids} 2>/dev/null || true
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)